### PR TITLE
[#176536665] New index page with e-mail field

### PR DESCRIPTION
**Pull request by vito80ba**
This PR create a new index page with a new field to store user's e-mail.
With this change the page with credit card fields are switched in a second screen.

#### List of Changes
- Create a new page
- Switched the credit card fields in a second screen
- Refactoring of some helpers in "real" typescript
- Modified pupeteer tests

#### Motivation and Context

The new flow designed by Jacopo require a specific page to store user's email.

#### How Has This Been Tested?

Pupeteer test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -12,274 +10,79 @@ document.addEventListener('DOMContentLoaded', () => {
...
-        () => undefined, // to be replaced with logic to handle failures
-        () => {
-          creditcardformInputs?.forEach(el => {
-            sessionStorage.setItem(el.getAttribute('name')?.trim() || '', el.value);
-          });
-          window.location.replace('check.html');
-        },
-      );
-      /*
-      e.preventDefault();
```
Comment (pasqualespica) [inline src/index.ts:181]: code comment if useless remove it

File: src/index.ts [hunk 2]
```diff
@@ -12,274 +10,79 @@ document.addEventListener('DOMContentLoaded', () => {
...
-    if (privacyTogglerInput.hasAttribute('checked') === true) {
-      privacyTogglerInput.removeAttribute('checked');
-      privacyTogglerInput.removeAttribute('data-checked');
+  // check if all fields are OK
+  function fieldsCheck() {
+    const checkedFields = emailform?.querySelectorAll('input[data-checked]');
+    if (checkedFields?.length === emailformInputs?.length) {
+      emailformSubmit?.removeAttribute('disabled');
     } else {
-      privacyTogglerInput.setAttribute('checked', '1'); // TODO: should be bool
```
Comment (pasqualespica) [inline src/index.ts:131]: Why this comment `bool`?

**Review by infantesimone — APPROVED** [review]
Well done 👍

File: src/inputcard.ts [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+  privacyToggler?.addEventListener('click', function () {
+    if (privacyTogglerInput == null) {
+      return;
+    }
+
+    if (privacyTogglerInput.hasAttribute('checked') === true) {
+      privacyTogglerInput.removeAttribute('checked');
+      privacyTogglerInput.removeAttribute('data-checked');
+    } else {
+      privacyTogglerInput.setAttribute('checked', '1'); // TODO: should be bool
```
Comment (pasqualespica) [inline src/inputcard.ts:131]: ditto

**Review by pasqualespica — APPROVED** [review]
Great work 👍

**Pull request merged**